### PR TITLE
Fix pom to make it compatible with Java 11

### PR DESCRIPTION
**Pull request by chtyim**
- Excludes the JDK tools.jar dependency from HBase. CDAP doesn't need the tools.jar
- Use Scala patch versions that are compatible with Java 11 for compilation

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -156,8 +156,8 @@
     <quartz.version>2.2.0</quartz.version>
     <resteasy.version>3.0.8.Final</resteasy.version>
     <rs-api.version>2.0</rs-api.version>
-    <scala2.10.version>2.10.4</scala2.10.version>
-    <scala2.11.version>2.11.8</scala2.11.version>
+    <scala2.10.version>2.10.7</scala2.10.version>
+    <scala2.11.version>2.11.12</scala2.11.version>
```
Comment (seanzhougoogle) [inline pom.xml:160]: what's the purpose of those version changes ? old version doesn't support Java 11 ?

File: pom.xml [hunk 2]
```diff
@@ -859,6 +863,10 @@
         <artifactId>hbase-client</artifactId>
         <version>${hbase10.version}</version>
         <exclusions>
+          <exclusion>
```
Comment (seanzhougoogle) [inline pom.xml:877]: when I did a dependency analysis, I found only hbase-common and tez-api depends on tools.jar. Do we need to exclude in other dependencies ?

File: pom.xml [hunk 3]
```diff
@@ -848,6 +848,10 @@
         <artifactId>hbase-common</artifactId>
         <version>${hbase10.version}</version>
         <exclusions>
+          <exclusion>
```
Comment (seanzhougoogle) [inline pom.xml:862]: so we dont' need to have any replacement? excluding it will just work ?

File: pom.xml (hunk 1 above)
Comment (chtyim) [reply pom.xml:160]: Right. Older Scala version will have an error during the compilation time `scala.reflect.internal.MissingRequirementError: object java.lang.Object in compiler mirror not found`.

Comment (seanzhougoogle) [reply pom.xml:160]: thanks for the information :)

File: pom.xml (hunk 2 above)
Comment (chtyim) [reply pom.xml:877]: Once you exclude it from hbase-common, try to run `dependency:tree` again, you'll see the dependency will get pulled in by some other hbase-* modules.

File: pom.xml (hunk 3 above)
Comment (chtyim) [reply pom.xml:862]: yes. We don't need the tools jar to build CDAP.

File: pom.xml (hunk 2 above)
Comment (seanzhougoogle) [reply pom.xml:877]: got it, thank you:)

File: pom.xml (hunk 3 above)
Comment (seanzhougoogle) [reply pom.xml:862]: got it, thank you :)

**Review by seanzhougoogle — APPROVED** [review]
Thank you, Terence :)

**Comment by chtyim** [comment]
There is more to make code actually can be run with a JRE 11. Mainly due to two changes since Java 9.

- Remove of the `sun.boot.class.path` property
- Introduction of the `jrt://` protocol for resources from classpath
-- The underlying Twill library needs to be upgraded to support the `jrt` protocol when tracing dependencies.

**Pull request closed without merging**